k8s/k8s.go

```diff
@@ -930,6 +934,15 @@ func (k *k8sOps) DescribeDeployment(depName, depNamespace string) (*apps_api.Dep
 	return &dep.Status, err
 }
 
+func (k *k8sOps) UpdateDeployment(deployment *apps_api.Deployment) (*apps_api.Deployment, error) {
+	logrus.Info("Attempt to update the deployment.")
+	dep, err := k.appsClient().Deployments(deployment.Namespace).Update(deployment)
```

Comment:
initialize the client using initK8sClient() before you call the Update here.. refer GetDeployment()
This will fail if this is the first call and the client is not initialized